dftool

```diff
@@ -911,6 +909,48 @@ Usage: dftool dispense_passive CHAINID AMOUNT
     print("Dispensed passive rewards")
 
 
+# ========================================================================
+@enforce_types
+def do_calculate_passive():
+    HELP = f"""Calculate passive rewards.
+
+Usage: dftool calculate_passive CHAINID AMOUNT DATE CSV_DIR
+    CHAINID -- {CHAINID_EXAMPLES}
+    DATE -- date in format YYYY-MM-DD
+    CSV_DIR -- output dir for passive-CHAINID.csv
+"""
+    if len(sys.argv) not in [6]:
+        print(HELP)
+        sys.exit(0)
+
+    CHAINID = int(sys.argv[2])
+    networkutil.connect(CHAINID)
+    DATE = sys.argv[4]
+    CSV_DIR = sys.argv[5]
+    timestamp = int(timestrToTimestamp(DATE))
+    S_PER_WEEK = 7 * 86400
+    timestamp = timestamp // S_PER_WEEK * S_PER_WEEK
+    ADDRESS_FILE = _getAddressEnvvarOrExit()
+    recordDeployedContracts(ADDRESS_FILE)
+
+    # load vebals csv file
+    passive_fname = csvs.passiveCsvFilename(CSV_DIR)
+    vebals_realtime_fname = csvs.vebalsCsvFilename(CSV_DIR, False)
+    if not os.path.exists(vebals_realtime_fname):
+        print(f"\nNo file {vebals_realtime_fname} in '{CSV_DIR}'. Exiting.")
+        sys.exit(0)
+    _exitIfFileExists(passive_fname)
+
+    # get addresses
+    vebals, _, _ = csvs.loadVebalsCsv(CSV_DIR, False)
+    addresses = list(vebals.keys())
+
+    balances, rewards = query.queryPassiveRewards(CHAINID, timestamp, addresses)
+
+    # save to csv
+    csvs.savePassiveCsv(rewards, balances, CSV_DIR)
```

Comment:
:+1: 

@alexcos20 are you using this data to feed df-sql like the rest of the pipeline?

Reply:
yeap